db_test.go

```diff
@@ -198,7 +202,59 @@ func TestDB_LowerBound_ByRevision(t *testing.T) {
 	require.EqualValues(t, 71, obj.ID)
 	_, _, ok = iter.Next()
 	require.False(t, ok)
+}
 
+func TestDB_Prefix(t *testing.T) {
+	t.Parallel()
+
+	db, table := newTestDBWithMetrics(t, &NopMetrics{}, tagsIndex)
+
+	{
+		txn := db.WriteTxn(table)
+		table.Insert(txn, testObject{ID: 42, Tags: []string{"a", "b"}})
+		table.Insert(txn, testObject{ID: 82, Tags: []string{"abc"}})
+		table.Insert(txn, testObject{ID: 71, Tags: []string{"ab"}})
+		txn.Commit()
+	}
+
+	txn := db.ReadTxn()
+
+	iter, watch := table.Prefix(txn, tagsIndex.Query("ab"))
+	require.Equal(t, Collect(Map(iter, testObject.getID)), []uint64{71, 82})
+
+	select {
+	case <-watch:
+		t.Fatalf("expected Prefix watch to not be closed before any changes")
+	default:
+	}
+
+	{
+		txn := db.WriteTxn(table)
+		table.Insert(txn, testObject{ID: 12, Tags: []string{"bc"}})
+		txn.Commit()
+	}
+
+	select {
+	case <-watch:
+		t.Fatalf("expected Prefix watch to not be closed before relevant changes")
+	default:
+	}
+
+	{
+		txn := db.WriteTxn(table)
+		table.Insert(txn, testObject{ID: 99, Tags: []string{"abcd"}})
+		txn.Commit()
+	}
+
+	select {
+	case <-watch:
+	case <-time.After(time.Second):
```

Comment:
If the CI machine skips a second due to virt hiccup or something, this becomes 50/50 afaik, since select choses from available channels pseudorandomly. Could check that `watch` isn't also closed before fatalling, even though that is also racy.

Reply:
I'll bump the timeouts.